### ISSUE TITLE
Clarify shutdown logic.

### ISSUE THREAD

**Issue by evanlinjin** [skycoin/skywire-testnet#327]
## Background
Due to the way `router.Serve()` is defined (with event loops in go-routines that start event loops in more go-routines), shutdown logic is unclear and hard to follow. The following steps are to clarify these processes.

Extracts from conversations on Discord:

> tpm.Serve() should be move out. Yes.
calling router.New() has TransportManager as input, one should have run tpm.Serve() beforehand.
We should have RoutingTableManager as an input for router.New(), instead of RoutingTable. rtm.Run() should also be run beforehand.

> From my perspective, we can just ensure that we shutdown the components in the right order.
Each component has a .Close() function.
In skywire/cmd/skywire-node/commands/root.go we should call .Close() in the right order.
In regards to how we deal with panics... from my understanding, that will be a topic of discussion.
I'm going to create a channel in the main group for this conversation.

**Comment by jdknives** [comment]
Closed by #376
